### PR TITLE
feat(aws): Add new check to ensure Aurora MySQL DB Clusters publish audit logs to CloudWatch logs

### PR DESCRIPTION
**Pull request by danibarranqueroo**
### Context

This new check verifies that Amazon Aurora MySQL DB clusters are configured to publish audit logs to Amazon CloudWatch Logs. Audit logs are vital for tracking database activity such as login attempts, data modifications, and schema changes. Configuring audit logs to publish to CloudWatch enables real-time analysis, durable storage of logs, and the creation of alarms and metrics for enhanced monitoring and compliance.

I have done this new check following the SH implementation, but there is an option of doing it as it’s done for instances, the option done is mirroring the existing AWS Security Hub control, focusing only on Aurora MySQL clusters, which have specific and robust support for audit log exports.
The other option would be doing as the [RDS.9](https://docs.aws.amazon.com/securityhub/latest/userguide/rds-controls.html#rds-9) control but for clusters, the problem here is that some clusters needs plugins to publish audit and they are not as complete as the Aurora MySQL ones.

So, in conclusion, I've done this check only for the Aurora MySQL clusters but the option of changing it is available and I’ll be open to listen any suggestion.

### Description

Added new `rds_cluster_integration_cloudwatch_logs` check with its unit tests.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        for db_cluster_arn, db_cluster in rds_client.db_clusters.items():
+            if db_cluster.engine == "aurora-mysql":
+                report = Check_Report_AWS(self.metadata())
+                report.region = db_cluster.region
+                report.resource_id = db_cluster.id
+                report.resource_arn = db_cluster_arn
+                report.resource_tags = db_cluster.tags
+                if db_cluster.cloudwatch_logs:
+                    report.status = "PASS"
+                    report.status_extended = f"Aurora MySQL Cluster {db_cluster.id} is shipping {' '.join(db_cluster.cloudwatch_logs)} to CloudWatch Logs."
```
Comment (sergargar) [inline prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py:17]:
```suggestion
                    report.status_extended = f"Aurora MySQL Cluster {db_cluster.id} is shipping {', '.join(db_cluster.cloudwatch_logs)} logs to CloudWatch Logs."
```

File: prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from prowler.lib.check.models import Check, Check_Report_AWS
+from prowler.providers.aws.services.rds.rds_client import rds_client
+
+
+class rds_cluster_integration_cloudwatch_logs(Check):
+    def execute(self):
+        findings = []
+        for db_cluster_arn, db_cluster in rds_client.db_clusters.items():
+            if db_cluster.engine == "aurora-mysql":
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py]: According to https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBCluster.html, it is valid for:
```
Valid for Cluster Type: Aurora DB clusters and Multi-AZ DB clusters

The following values are valid for each DB engine:

Aurora MySQL - audit | error | general | slowquery

Aurora PostgreSQL - postgresql

RDS for MySQL - error | general | slowquery

RDS for PostgreSQL - postgresql | upgrade
```

Please, make the check look for these engines too.

File: prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+                report = Check_Report_AWS(self.metadata())
+                report.region = db_cluster.region
+                report.resource_id = db_cluster.id
+                report.resource_arn = db_cluster_arn
+                report.resource_tags = db_cluster.tags
+                if db_cluster.cloudwatch_logs:
+                    report.status = "PASS"
+                    report.status_extended = f"Aurora MySQL Cluster {db_cluster.id} is shipping {', '.join(db_cluster.cloudwatch_logs)} logs to CloudWatch Logs."
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py]:
```suggestion
                    report.status_extended = f"RDS Cluster {db_cluster.id} is shipping {', '.join(db_cluster.cloudwatch_logs)} logs to CloudWatch Logs."
```

File: prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+                else:
+                    report.status = "FAIL"
+                    report.status_extended = f"Aurora MySQL Cluster {db_cluster.id} does not have CloudWatch Logs enabled."
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py]:
```suggestion
                    report.status_extended = f"RDS Cluster {db_cluster.id} does not have CloudWatch Logs enabled."
```

File: prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py (hunk 2 above)
Comment (danibarranqueroo) [reply prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py]: Done!

File: prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py (hunk 3 above)
Comment (danibarranqueroo) [reply prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py]: Done.

File: prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py (hunk 4 above)
Comment (danibarranqueroo) [reply prowler/providers/aws/services/rds/rds_cluster_integration_cloudwatch_logs/rds_cluster_integration_cloudwatch_logs.py]: Done.

**Review by MrCloudSec — APPROVED** [review]
👏🏼
